Core/Operator/Symbol.cs

```diff
@@ -460,49 +463,45 @@ public Instance CreateInstance(Guid id)
 
             SortInputSlotsByDefinitionOrder(newInstance.Inputs);
 
-            // create child instances
-            foreach (var symbolChild in Children)
+            foreach (var child in Children)
             {
-                CreateAndAddNewChildInstance(symbolChild, newInstance);
+                CreateAndAddNewChildInstance(child, newInstance);
             }
 
             // create connections between instances
-            var conHashToCount = new Dictionary<ulong, int>(Connections.Count);
-            var invalidConnections = new List<Connection>();
-            foreach (var connection in Connections)
-            {
-                ulong highPart = 0xFFFFFFFF & (ulong)connection.TargetSlotId.GetHashCode();
-                ulong lowPart = 0xFFFFFFFF & (ulong)connection.TargetParentOrChildId.GetHashCode();
-                ulong hash = (highPart << 32) | lowPart;
-                if (!conHashToCount.TryGetValue(hash, out int count))
-                    conHashToCount.Add(hash, 0);
-
-                var valid = newInstance.AddConnection(connection, count);
-                if (!valid)
+            if (Connections.Count != 0)
+            {
+                var conHashToCount = new Dictionary<ulong, int>(Connections.Count);
+                for (var index = 0; index < Connections.Count; index++) // warning: the order in which these are processed matters
                 {
-                    Log.Warning("Skipping connection to no longer existing targets");
-                    invalidConnections.Add(connection);
-                    continue;
-                }
- 
-                conHashToCount[hash] = count + 1;
-            }
+                    var connection = Connections[index];
+                    ulong highPart = 0xFFFFFFFF & (ulong)connection.TargetSlotId.GetHashCode();
+                    ulong lowPart = 0xFFFFFFFF & (ulong)connection.TargetParentOrChildId.GetHashCode();
+                    ulong hash = (highPart << 32) | lowPart;
+                    if (!conHashToCount.TryGetValue(hash, out int count))
+                        conHashToCount.Add(hash, 0);
+
+                    var valid = newInstance.AddConnection(connection, count);
+                    if (!valid)
+                    {
+                        Log.Warning("Skipping connection to no longer existing targets");
+                        Connections.RemoveAt(index);
+                        index--;
+                        continue;
+                    }
 
-            foreach (var c in invalidConnections)
-            {
-                Connections.Remove(c);
+                    conHashToCount[hash] = count + 1;
+                }
```

Comment:
potentially removed several iterations over the list by removing connections from within the loop itself.

fun fact: this is where I broke everything :) took me way too long to realize... hence the comment lmao

Reply:
Okay